app/src/main/java/io/homeassistant/companion/android/sensors/SensorReceiver.kt

```diff
@@ -151,6 +152,16 @@ class SensorReceiver : BroadcastReceiver() {
         val sensorDao = AppDatabase.getInstance(context).sensorDao()
         val enabledRegistrations = mutableListOf<SensorRegistration<Any>>()
 
+        val checkDeviceRegistration = integrationUseCase.getRegistration()
+        if (checkDeviceRegistration.appVersion == null) {
+            Log.w(TAG, "Device not registered, skipping sensor update/registration")
+            return
+        }
+
+        integrationUseCase.setInstalledAppVersion(BuildConfig.VERSION_NAME)
```

Comment:
Why do we need to store this in the preference?

Reply:
Umm wasn't sure if we would need it in another place. I can remove and check build config directly